nemo/collections/asr/models/asr_model.py

```diff
@@ -173,51 +173,82 @@ def on_after_backward(self):
                 logging.warning('detected inf or nan values in gradients! Setting gradients to zero.')
                 self.zero_grad()
 
+    def maybe_enable_cuda_graphs(self, force_reinit=False) -> bool:
+        """Enable CUDA graphs if all conditions met, return True if state changed"""
+        state_changed = False
+        if force_reinit:
+            state_changed = self.disable_cuda_graphs()
+        # check that self.decoding.decoding exists and is instance of WithOptionalCudaGraphs
+        if isinstance(getattr(getattr(self, "decoding", None), "decoding", None), WithOptionalCudaGraphs):
+            state_changed |= self.decoding.decoding.maybe_enable_cuda_graphs()
+        if state_changed:
+            logging.info(
+                f"CUDA graphs enabled for {type(self).__name__}::{type(self.decoding).__name__}::"
+                f"{type(self.decoding.decoding).__name__}"
+            )
+        return state_changed
+
+    def disable_cuda_graphs(self) -> bool:
+        """Disable (maybe temporary) CUDA graphs, return True if state changed"""
+        state_changed = False
+        # check that self.decoding.decoding exists and is instance of WithOptionalCudaGraphs
+        if isinstance(getattr(getattr(self, "decoding", None), "decoding", None), WithOptionalCudaGraphs):
+            state_changed = self.decoding.decoding.disable_cuda_graphs()
+        if state_changed:
+            logging.info(
+                f"CUDA graphs disabled for {type(self).__name__}::{type(self.decoding).__name__}::"
+                f"{type(self.decoding.decoding).__name__}"
+            )
+        return state_changed
+
     def on_train_epoch_start(self) -> None:
         """
         Decoder with CUDA graphs does not release memory, thus we disable it for training epoch.
         EncDecRNNTModel.decoding.decoding is the inference class with CUDA graphs
         """
-        WithOptionalCudaGraphs.disable_cuda_graphs_recursive(self, attribute_path="decoding.decoding")
+        self.disable_cuda_graphs()
 
     def on_train_epoch_end(self) -> None:
         """
         After training, we can enable the decoder with CUDA graphs.
         EncDecRNNTModel.decoding.decoding is the inference class with CUDA graphs
         """
-        WithOptionalCudaGraphs.enable_cuda_graphs_recursive(self, attribute_path="decoding.decoding")
+        self.maybe_enable_cuda_graphs()
 
     def on_validation_epoch_start(self) -> None:
         """
         For validation, we enable CUDA graphs to speedup validation.
+        Force re-enabling required due to issues with trainer and mixed precision.
```

Comment:
Is this happening with any specific version?

Should we comeback and remove when fix happens with latest versions?